Code/MinimalLib/minilib.cpp

```diff
@@ -954,3 +954,47 @@ JSLog *set_log_capture(const std::string &log_name) {
 void enable_logging() { RDKit::MinimalLib::LogHandle::enableLogging(); }
 
 void disable_logging() { RDKit::MinimalLib::LogHandle::disableLogging(); }
+
+#ifdef RDK_BUILD_MINIMAL_LIB_RGROUPDECOMP
+int JSRgroupDecomp::add(const JSMol &mol) {
+  return decomp.add(RDKit::RWMol(*mol.d_mol));
+}
+
+bool JSRgroupDecomp::process() {
+  return decomp.process();
+}
+
+std::pair<std::vector<std::string>, std::vector<JSMolList*>> JSRgroupDecomp::getRGroupsAsColumns() const {
+  RGroupColumns cols = decomp.getRGroupsAsColumns();
+
+  std::vector<std::string> keys;
+  for (std::map<std::string, RGroupColumn>::iterator it = cols.begin(); it != cols.end(); ++it)
+    keys.push_back(it->first);
+  
+  std::vector<JSMolList*> res(keys.size());
+  for (size_t i = 0; i < keys.size(); ++i)
```

Comment:
Is it possible to use one for loop instead of 2 separate? I assume you are using the second one because in this case you do not need to use push and can allocate the array in advance? But is there a way to know the size of cols? And just use it to allocate memory for both keys and res and use one loop to fill both arrays?

Reply:
will do